src/OrchardCore.Modules/OrchardCore.Sms.Azure/Services/AzureSmsProviderBase.cs

```diff
@@ -0,0 +1,75 @@
+using Azure.Communication.Sms;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Sms.Azure.Models;
+
+namespace OrchardCore.Sms.Azure.Services;
+
+public abstract class AzureSmsProviderBase : ISmsProvider
+{
+    private readonly AzureSmsOptions _providerOptions;
+    private readonly IPhoneFormatValidator _phoneFormatValidator;
+    private readonly ILogger _logger;
+
+    protected readonly IStringLocalizer S;
+
+    public AzureSmsProviderBase(
+        AzureSmsOptions options,
+        IPhoneFormatValidator phoneFormatValidator,
+        ILogger logger,
+        IStringLocalizer stringLocalizer)
+    {
+        _providerOptions = options;
+        _phoneFormatValidator = phoneFormatValidator;
+        _logger = logger;
+        S = stringLocalizer;
+    }
+
+    public abstract LocalizedString Name { get; }
+
+    public virtual async Task<SmsResult> SendAsync(SmsMessage message)
+    {
+        ArgumentNullException.ThrowIfNull(message);
+
+        if (!_providerOptions.IsEnabled)
+        {
+            return SmsResult.Failed(S["The Azure SMS Provider is disabled."]);
+        }
+
+        _logger.LogDebug("Attempting to send SMS to {Recipient}.", message.To);
+
+        if (string.IsNullOrWhiteSpace(message.To))
+        {
+            return SmsResult.Failed(S["A phone number is required for the recipient.", message.To]);
+        }
+
+        if (!_phoneFormatValidator.IsValid(message.To))
+        {
+            return SmsResult.Failed(S["Invalid phone number format for the recipient: '{0}'.", message.To]);
+        }
+
+        if (string.IsNullOrEmpty(message.Body))
+        {
+            return SmsResult.Failed(S["The message body is required.", message.To]);
+        }
+
+        try
+        {
+            var client = new SmsClient(_providerOptions.ConnectionString);
```

Comment:
Store this instance in the class so if the Send method was called multiple times with in the same request, we don't create a new instance every time